### PR TITLE
fix: limit the amount of auctions being loaded at once 

### PR DESCRIPTION
**Pull request by 0xArdy**
- fix: limit the amount of auctions being loaded at once
- add deploy workflow
- add build+test workflow
- add eslint
- update yarn.lock

closes HAI-321

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "dependencies": {
         "@ethersproject/experimental": "5.4.0",
         "@ethersproject/providers": "5.4.5",
-        "@hai-on-op/sdk": "1.2.0",
+        "@hai-on-op/sdk": "1.2.0-99dc1efc.0",
```
Comment (0xGorilla) [inline package.json]: let's merge the sdk before merging this PR

Comment (wei3erHase) [reply package.json]: @0xArdy i understand 1.2.1 is already live

Comment (0xArdy) [reply package.json]: [not yet](https://www.npmjs.com/package/@hai-on-op/sdk?activeTab=versions), we need to merge this pr first  https://github.com/hai-on-op/sdk/pull/6

**Comment by 0xArdy** [comment]
@wei3erHase @0xGorilla should I remove the vercel config files? I think it will break the current `hai-on-op/app` vercel configuration

File: package.json [hunk 2]
```diff
@@ -66,27 +66,30 @@
...
+        "eslint-config-prettier": "9.0.0",
+        "eslint-plugin-import": "2.28.1",
+        "eslint-plugin-jsx-a11y": "6.7.1",
+        "eslint-plugin-prettier": "5.0.0",
+        "eslint-plugin-react": "7.33.2",
+        "eslint-plugin-react-hooks": "4.6.0",
+        "eslint-plugin-react-refresh": "0.4.3",
         "husky": "^5.0.9",
         "jsdom": "22.1.0",
         "lint-staged": "^10.5.4",
```
Comment (turtlemoji) [inline package.json]: Remove carets and put last version from lock

File: src/utils/constants.ts [hunk 3]
```diff
@@ -56,3 +56,9 @@ export const floatsTypes = {
 }
 
 export const network_name = VITE_NETWORK_ID === '1' ? 'mainnet' : 'optimism-goerli'
+
+// Auctions
+export const NUMBER_OF_AUCTIONS_TO_SHOW = 15
+export const SURPLUS_BATCH_SIZE = 5000000 // blocks
```
Comment (turtlemoji) [inline src/utils/constants.ts]: Maybe "5_000_000" would be easier to read

File: src/utils/getAuctions.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    // Ensure that startBlock is never smaller than endBlock.
+    if (endBlock <= startBlock) {
+        return { auctions: accumulatedAuctions, endBlock: startBlock - 1 }
+    }
+
+    const auctionsFetched = await fetchFunction(startBlock, endBlock)
+    const totalAuctions = accumulatedAuctions.concat(auctionsFetched.auctions)
+
+    if (totalAuctions.length || endBlock <= startBlock) {
```
Comment (turtlemoji) [inline src/utils/getAuctions.ts]: The code "endBlock <= startBlock" will never be reached. Check line 22

File: src/hooks/useLoadMoreAuctions.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+            )}
+        </BtnContainer>
+    )
+
+    return { LoadMoreButton, numberOfAuctions }
+}
+
+export const BtnContainer = styled.div`
+    text-align: right;
+    padding-top: 15px;
```
Comment (turtlemoji) [inline src/hooks/useLoadMoreAuctions.tsx:99]: Not rem on hai?

File: src/hooks/useLoadMoreAuctions.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+}
+
+export const useLoadMoreAuctions = ({ auctions }: Props) => {
+    const geb = usePublicGeb()
+    const { auctionModel: auctionsActions } = useStoreActions((state) => state)
+    const { auctionModel: auctionsState, connectWalletModel: connectWalletState } = useStoreState((state) => state)
+    const [numberOfAuctions, setNumberOfAuctions] = useState(NUMBER_OF_AUCTIONS_TO_SHOW)
+    const { proxyAddress } = connectWalletState
+    const { loadingAuctionsData } = auctionsState
+    const type = _.get(auctions?.[0], 'englishAuctionType', undefined)
```
Comment (turtlemoji) [inline src/hooks/useLoadMoreAuctions.tsx]: Using underscore as lib name is weird in this lines

File: src/utils/formatAuctions.ts [hunk 7]
```diff
@@ -0,0 +1,187 @@
+import { BigNumber, utils } from 'ethers'
+import { ISurplusAuction as SDKAuction, ICollateralAuction, utils as gebUtils } from '@hai-on-op/sdk'
+
+import _ from '~/utils/lodash'
```
Comment (turtlemoji) [inline src/utils/formatAuctions.ts]: Ídem

File: src/utils/formatAuctions.ts [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+    if (auctionsList) {
+        if (auctionsList.length === 0) {
+            return []
+        }
+
+        // show auctions less than one month old only
+        // const oneMonthOld = new Date().setMonth(new Date().getMonth() - 1)
+        const filteredAuctions: IAuction[] = auctionsList.map((auc: SDKAuction) => {
+            const {
+                // isClaimed,
```
Comment (turtlemoji) [inline src/utils/formatAuctions.ts]: We need the comments?

File: src/utils/formatAuctions.ts [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+
+export const formatSurplusAndDebtAuctions = (auctionsList: SDKAuction[], userProxy: string): IAuction[] => {
+    if (auctionsList) {
+        if (auctionsList.length === 0) {
+            return []
+        }
+
+        // show auctions less than one month old only
+        // const oneMonthOld = new Date().setMonth(new Date().getMonth() - 1)
+        const filteredAuctions: IAuction[] = auctionsList.map((auc: SDKAuction) => {
```
Comment (turtlemoji) [inline src/utils/formatAuctions.ts]: auction would be more specific than auc without repercussions

File: src/utils/formatAuctions.ts [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+    }
+}
+
+export const formatCollateralAuctions = (auctionsList: any[], tokenSymbol: string): ICollateralAuction[] => {
+    if (auctionsList) {
+        if (auctionsList.length === 0) {
+            return []
+        }
+
+        const filteredAuctions = auctionsList.map((auc: ICollateralAuction, index) => {
```
Comment (turtlemoji) [inline src/utils/formatAuctions.ts]: Ídem about auction name

File: src/utils/formatAuctions.ts [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+            const { createdAt, createdAtTransaction, amountToSell, amountToRaise, biddersList } = auc
+
+            const startedBy = _.get(auc, 'startedBy', '')
+            // Amount to sell = collateral
+            // Amout to raise = hai
+            const collateralBought = biddersList.reduce((acc, bid) => acc.add(bid.bid), BigNumber.from('0'))
```
Comment (turtlemoji) [inline src/utils/formatAuctions.ts]: acc as account? Ídem about it

File: src/hooks/useLoadMoreAuctions.tsx (hunk 5 above)
Comment (0xArdy) [reply src/hooks/useLoadMoreAuctions.tsx:99]: nope

File: src/utils/constants.ts (hunk 3 above)
Comment (0xArdy) [reply src/utils/constants.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/de69465500f3b003e69cae07368995be9ed7beea

File: src/utils/getAuctions.ts (hunk 4 above)
Comment (0xArdy) [reply src/utils/getAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/9cfb83d77c541ae6c3dfd081dd6ad27331dea34b

File: src/utils/formatAuctions.ts (hunk 7 above)
Comment (0xArdy) [reply src/utils/formatAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/38b84018a9d4540bc62426a0ee34f45b47545547

File: src/hooks/useLoadMoreAuctions.tsx (hunk 6 above)
Comment (0xArdy) [reply src/hooks/useLoadMoreAuctions.tsx]: https://github.com/defi-wonderland/hai-app/pull/2/commits/38b84018a9d4540bc62426a0ee34f45b47545547

File: src/utils/formatAuctions.ts (hunk 11 above)
Comment (0xArdy) [reply src/utils/formatAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/fb0aff099d5c474546da3b80f63db7b5cfad5eba

File: src/utils/formatAuctions.ts (hunk 9 above)
Comment (0xArdy) [reply src/utils/formatAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/e4440ab5fc48afea04ba94e787645ff6eb489ca0

File: src/utils/formatAuctions.ts (hunk 10 above)
Comment (0xArdy) [reply src/utils/formatAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/e4440ab5fc48afea04ba94e787645ff6eb489ca0

File: src/utils/formatAuctions.ts (hunk 8 above)
Comment (0xArdy) [reply src/utils/formatAuctions.ts]: https://github.com/defi-wonderland/hai-app/pull/2/commits/abe8e0f0eef99e0335a00bcd66e38c2dc501a738

File: package.json (hunk 2 above)
Comment (0xArdy) [reply package.json]: https://github.com/defi-wonderland/hai-app/pull/2/commits/3c86172c2f9d7794c8682398d6f13abed4f500a7

File: src/utils/constants.ts (hunk 3 above)
Comment (turtlemoji) [reply src/utils/constants.ts]: But can you modify the other number constants too?

Comment (0xArdy) [reply src/utils/constants.ts]: done https://github.com/defi-wonderland/hai-app/pull/2/commits/204e1978ef637cf439bc548d560251dec8422e4b

**Review by turtlemoji — APPROVED** [review]

**Pull request merged**